### PR TITLE
LPS-200407 | Refactor all implementations of Autocomplete component to use SingleSelect

### PR DESCRIPTION
**Pull request by andreefarias**
**Ticket: [LPS-200407](https://liferay.atlassian.net/browse/LPS-200407)**
**Original PR: #2820**

## Refactor:
* Refactor the select structure to accessibility using the framework clay on objects, picklists, notifications
* Refactor the testcases related from this.

## Testcases fixed: 
* [ci:test:object](https://testray.liferay.com/home/-/testray/subtasks/index?delta=200&p_p_state=normal&orderByCol=score_sortable&orderByType=desc&testrayTaskId=2455869347)

## Screenshot (exemple):
image

### CODE REVIEW

**Comment by andreefarias** [comment]
ci:test:object

**Comment by andreefarias** [comment]
ci:test:relevant

**Comment by andreefarias** [comment]
I sent a new PR to run the CI tests suites with the new changes related the testcases.

Original PR: #2820

cc: @danvb112 @rebsilva

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsDetailsDisplayContext.java [hunk 1]
```diff
@@ -173,20 +172,29 @@ public List<KeyValuePair> getScopeKeyValuePairs(String scope) {
 				_panelCategoryRegistry.getChildPanelCategories(
 					panelCategoryKey);
 
+			JSONArray itensJSONArray = JSONFactoryUtil.createJSONArray();
```
Comment (marcelabc) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsDetailsDisplayContext.java:175]: We should rename this variable to ite**m**sJSONArray 👍

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-test/src/testFunctional/tests/Picklist.testcase [hunk 2]
```diff
@@ -1021,10 +1021,10 @@ definition {
 			ObjectAdmin.selectDropdownItem(
 				currentOption = "Select an option",
 				labelName = "Tipo",
-				optionName = "Lista de seleção");
+				optionName = "Picklist");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/Picklist.testcase:1024]: @andreefarias is this correct expected behavior? The option name will no longer be translated?

File: modules/apps/object/object-test/src/testFunctional/tests/Picklist.testcase [hunk 3]
```diff
@@ -1021,10 +1021,10 @@ definition {
...
 
 			ObjectAdmin.selectDropdownItem(
-				currentOption = "Escolha uma Opção",
+				currentOption = "Select an option",
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/Picklist.testcase:1027]: Same thing here @andreefarias

File: modules/apps/object/object-test/src/testFunctional/tests/Picklist.testcase (hunk 2 above)
Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/Picklist.testcase:1024]: Yes, should be.

File: modules/apps/object/object-test/src/testFunctional/tests/Picklist.testcase (hunk 3 above)
Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/Picklist.testcase:1027]: Yes, should be.

**Comment by andreefarias** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:stop

**Comment by andreefarias** [comment]
ci:test:object

**Comment by andreefarias** [comment]
ci:test:relevant

**Comment by andreefarias** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by andreefarias** [comment]
ci:forward

**Comment by andreefarias** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
ci:forward:force

**Comment by andreefarias** [comment]
ci:forward:force

**Pull request closed without merging**